conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegration.java

```diff
@@ -285,24 +295,29 @@ public void resumeTracking() {
     }
 
     /**
-     * This should be called when the app is resumed
+     * This should be called when the app is resumed.
+     * <p>
+     * NOTE: This method must not be called, when the analytics is created via `ConvivaAnalytics.buildVideoAnalytics` or `ConvivaAnalytics.buildAdAnalytics` as this is handled internally already!
      */
     public void reportAppForegrounded() {
         Log.d(TAG, "appForegrounded");
-        ConvivaAnalytics.reportAppForegrounded();
-        isBackgrounded = false;
+        if (isBackgrounded) {
+            ConvivaAnalytics.reportAppForegrounded();
+            isBackgrounded = false;
+        }
     }
 
     /**
      * This should be called when the app is paused
+     * <p>
+     * NOTE: This method must not be called, when the analytics is created via `ConvivaAnalytics.buildVideoAnalytics` or `ConvivaAnalytics.buildAdAnalytics` as this is handled internally already!
```

Comment:
so the constructors that should be used are  only following 2?
- `public ConvivaAnalyticsIntegration(Player player, String customerKey, Context context)`
- `public ConvivaAnalyticsIntegration(Player player, String customerKey, Context context, ConvivaConfig config)`
what about the constructor with `ssai`? also for testing only?

in the case the other constructors are for testing only, I would mark them as `@VisibleForTesting`, to disallow customers the usage of them. (which would allow us to remove the functions again :thinking: )

Reply:
The are removed in https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/81, still I am not super happy with just removing the functions given that this might be in use by customers. 
I would suggest to keep them for now and maybe deprecate them to indicate that they should not be used any more. 
Maybe there is still a scenario out there that we are missing where they could be useful. @dweinber or @stonko1994 might know more.

Reply:
I didn't notice this PR :see_no_evil: -> will update and resolve the merge conflicts tomorrow first thing

ok, I am fine to keep the functions and mark them as deprecated :+1:

Reply:
I don't think there is any value in having those constructors. I looked up the commit where they were introduced (at least the first one): https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/commit/ce878481ae7979b4ebc4f6a1100940615cbf39c3#diff-4c62961b542a5646de9a27ab30562e6b508b8f3438a8c950a893960b81ee5733.

The commit messages clearly indicates that this was just for testing (mocking/stubbing). I'd take the risk of someone using it and keeping them removed.

Reply:
@stonko1994 thanks for your input about the constructors.
What do you think about the functions `reportAppForegrounded` and `...Backgrounded`? (these functions shouldn't be called anymore, as this is handled inside the conviva-core sdk and will produce error messages in touchstone when being called by the customer)

@strangesource I updated the comment and marked the functions as deprecated here: [19ad7c3](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/78/commits/19ad7c33e9107c05c3e8772cbaa5f6aa6d1a39f6)

Reply:
They were definitelly needed in the past and we also mention the need to call them in the readme AFAIK. If they should no longer used I'd deprecate them but I wouldn't throw errors (only log warnings).